index.rst

```diff
@@ -1,51 +1,1667 @@
+            start, stop, stride = index.indices(self.getSize())
+            if stride != 1:
+                raise IndexError("Non-unit stride not supported")
+            return self.slice(start, stop)
+        return self._get_(index)
+
+which uses the ``getSize``, ``slice``, and ``_get_`` methods defined in the pybind11 wrapper to
+generate a slice (if necessary).
+To make this the ``__getitem__`` method in ``ArrayFKey`` and ``ArrayIKey`` we add
+
+.. code-block:: python
+
+
+    ArrayFKey.__getitem__ = _getitem_
+    ArrayDKey.__getitem__ = _getitem_
+    del _getitem_
+
+which assigns the ``__getitem__`` method to the classes and deletes the temporary function so that
+it doesn't pollute the namespace.
+Finally we must add ``from .arrays import *`` to ``tableLib.py`` to ensure that the stack updates
+both classes. The complete ``arrays.py`` file should be
+
+.. code-block:: python
+
+    from __future__ import absolute_import, division, print_function
+    from ._arrays import ArrayFKey, ArrayDKey
+
+    def _getitem_(self, index):
+        """
+        operator[] in C++ only returns a single item, but `Array` has a method to get a slice of the
+        array. To make the code more python we automatically check for a slice and return either
+        a single item or slice as requested by the user.
+        """
+        if isinstance(index, slice):
+            start, stop, stride = index.indices(self.getSize())
+            if stride != 1:
+                raise IndexError("Non-unit stride not supported")
+            return self.slice(start, stop)
+        return self._get_(index)
+
+    ArrayFKey.__getitem__ = _getitem_
+    ArrayDKey.__getitem__ = _getitem_
+
+
+In most cases, the SWIG files from the current stack will contain the necessary python code and one can
+simply copy and paste the code from the SWIG file into the new python file with little modification.
+
+.. _fep:
+
+Frequently Encountered Problems
+===============================
+
+There are a number of errors, issues, and other problems that you are likely to come across during wrapping.
+This section has some hints on what might be causing a particular problem you are encountering.
+
+Casting
+-------
+
+SWIG and pybind11 handle inheritance in different ways. In SWIG, if a class B inherits from A,
+a pointer that clones B can return a type A, which is undesirable.
+There was a lot of machinery, including a ``.cast`` method that was used to recase A as B.
+This is not necessary with pybind11 so all casting procedures can be removed
+(or at the very least commented out) and tests for casting can be skipped with a 
+``@unittest.skip("Skip for pybind11")``.
+
+.. _segfaults:
+
+Segmentation Faults
+-------------------
+
+Smart Pointers
+^^^^^^^^^^^^^^
+
+The vast majority of the segfaults you encounter will be caused by inheriting a class that is defined
+with a smart pointer, but not using a smart pointer in the template definition of the new class
```

Comment:
But not using *the same* smart pointer -- pybind11 always associates  holder class, by default `std::unique_ptr`.